src/devices/cpu/m6502/m6502.h

```diff
@@ -136,8 +136,8 @@ class m6502_device : public cpu_device {
 	uint8_t read_arg(uint16_t adr) { return mintf->read_arg(adr); }
 	uint8_t read_pc() { return mintf->read_arg(PC++); }
 	uint8_t read_pc_noinc() { return mintf->read_arg(PC); }
-	void prefetch();
-	void prefetch_noirq();
+	virtual void prefetch();
```

Comment:
Don't do that.  Specialize the memory accessor instead and change read_sync/decrypt8.  See mame/machine/deco222.* for an example.